pkg/trace/agent/agent.go

```diff
@@ -304,7 +312,9 @@ func (a *Agent) Process(p *api.Payload) {
 		}
 
 		numEvents, keep, filteredChunk := a.sample(now, ts, pt)
-		if !keep {
+		if !keep && sampler.ApplySpanSampling(chunk) {
+			// The span sampler decided to keep some of the spans anyway. It modified chunk in-place.
+		} else if !keep {
```

Comment:
If `!keep && sampler.SingleSpans(chunk) == true`, then we don't want to enter the other block, but we do want to do what follows the block.

Reply:
I'll push an alternative. Let me know what you think.

Reply:
Pushed.

Another possibility is to change `ApplySpanSampling` so that it doesn't modify its argument in-place, but instead optionally returns a new thing.  That could have the advantage of removing a little mystery, and would be an opportunity to change the name of the function.

Let me know how that sounds.

My preference is to leave the empty `if` body.

Reply:
Oh, I didn't understand you suggestion.

That's too tricky...

Hm, let's see.

Reply:
Ok, have it your way.